lib/compilers/nim.js

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2019, Bastien Penavayre
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+const BaseCompiler = require('../base-compiler'),
+    _ = require('underscore'),
+    path = require('path'),
+    argumentParsers = require("./argument-parsers"),
+    fs = require('fs-extra');
+
+class NimCompiler extends BaseCompiler {
+    constructor(info, env) {
+        super(info, env);
+        this.compiler.supportsIntel = true;
+    }
+
+    cacheDir(outputFilename) {
+        return outputFilename + '.cache';
+    }
+
+    optionsForFilter(filters, outputFilename) {
+        return [
+            "-o:" + outputFilename, //output file, only for js mode
+            "--nolinking", //disable linking, only compile to nimcache
+            "--nimcache:" + this.cacheDir(outputFilename) //output folder for the nimcache
+        ];
+    }
+
+    filterUserOptions(userOptions) {
+        //Allowed commands
+        const commands = [
+            "compile", "compileToC", "c",
+            "compileToCpp", "cpp", "cc",
+            "compileToOC", "objc",
+            "js",
+            "check"
+        ];
+
+        //If none of the allowed commands is present in userOptions compile to C++
+        if (_.intersection(userOptions, commands).length === 0) {
```

Comment:
Maybe it's better to still use `C` as a default compiler target since that's the case in Nim itself? :thinking:

Reply:
Yap fair enough.